flink-dist/src/main/flink-bin/bin/taskmanager.sh

```diff
@@ -63,7 +57,7 @@ if [[ $STARTSTOP == "start" ]]; then
         #
         TM_MAX_OFFHEAP_SIZE="8388607T"
 
-        if [[ "${STREAMINGMODE}" == "batch" ]] && useOffHeapMemory; then
+        if [[ "${FLINK_TM_MEM_ALLOC}" == "eager" ]] && useOffHeapMemory; then
```

Comment:
are you sure? I just preserved the previous behavior. (eager == batch_mode, lazy == streaming_mode)